### PR TITLE
Change PodLogs test to use testClient validate

### PR DESCRIPTION
**Pull request by krousey**
And also switch the 200s to http.StatusOK while I'm here.

I'm mainly doing this because the test depended on accessing private member of the request object. And I'm trying to move that to another package.

### CODE REVIEW

**Comment by krousey** [comment]
@kubernetes/goog-csi

File: pkg/client/unversioned/pods_test.go [hunk 1]
```diff
@@ -194,22 +195,22 @@ func TestPodGetLogs(t *testing.T) {
...
+			Method: "GET",
+			Path:   testapi.Default.ResourcePath("pods", ns, "podName") + "/log",
+			Query: url.Values{
+				"follow":     []string{"true"},
+				"timestamps": []string{"true"},
+			},
+		},
+		Response: Response{StatusCode: http.StatusOK},
 	}
-	expected := map[string]string{"container": "", "follow": "true", "previous": "false", "timestamps": "true"}
```
Comment (caesarxuchao) [inline pkg/client/unversioned/pods_test.go]: Who's checking this after the change? ValidateCommon seems not checking the request.params.

Comment (krousey) [inline pkg/client/unversioned/pods_test.go]: https://github.com/kubernetes/kubernetes/blob/master/pkg/client/unversioned/client_test.go#L163

Comment (caesarxuchao) [inline pkg/client/unversioned/pods_test.go]: Sorry, I misunderstood the usage of the request in testClient. Nevermind my previous question.

Did you remove `"previous": "false"` intentionally?

Comment (krousey) [inline pkg/client/unversioned/pods_test.go]: Yes. testClient actually roundtrips a request to a local http server. In that round trip, empty and false query parameters get stripped, so the server never sees them.

Comment (caesarxuchao) [inline pkg/client/unversioned/pods_test.go]: Ok.

**Comment by caesarxuchao** [comment]
LGTM. Thank you.

**Pull request merged**